Doc/library/types.rst

```diff
@@ -82,6 +82,32 @@ Dynamic Type Creation
 
    .. versionadded:: 3.7
 
+.. function:: get_original_bases(cls, /)
+
+    Return the objects in the bases list of the class's definition as they
+    existed prior to any modification by :meth:`~object.__mro_entries__`. This
```

Comment:
What does "modification by `__mro_entries__`" refer to? In my mind that is a read-only function that doesn't modify anything.

But I'm probably missing another subtle distinction. :-)

Reply:
I still don't follow. Can you point me to the code where `__mro_entries__` modifies something?

FWIW In general it's better not to describe things in terms of "internals", but instead refer to the specification of an object or operation in the documentation. (And once something is documented it's no longer internal, even if it has a dunder name.)

Reply:
Okay, I have to admit I was again fooled by misreading. :-( I might as well come clear, I didn't remember the ugly PEP 560 mechanism and thought you were referencing `__mro__`. :-(

Maybe we can work a PEP 560 reference in the text, e.g. "Return the tuple of base classes originally passed to class creation, before they were replaced the PEP 560 mechanism using `__mro_entries__`."

I'm really sorry for being so dense.

Reply:
Hmm, well we already have an inline link to the `__mro_entries__` docs, which I just revamped the other day in https://github.com/python/cpython/pull/103374 and https://github.com/python/cpython/pull/103398. I _hope_ they describe the process pretty well now: https://docs.python.org/3.11/reference/datamodel.html#object.__mro_entries__

I guess a PEP-560 link as well wouldn't hurt, though I thought we were trying to move away from using PEPs as canonical sources of documentation :)

Reply:
Well, you did ask me for a review. :-) PEP 560 is one of the most obscure mechanisms in all of typing, so deserves a mention whenever it is used. Also, perhaps a mention that this essentially just returns `__orig_bases__`, if it exists, might be helpful.

Reply:
Sure, let's add the PEP-560 link then. Though I don't know how helpful mentioning `__orig_bases__` would be for most users, since the attribute is currently completely undocumented AFAIK?

Reply:
> Though I don't know how helpful mentioning `__orig_bases__` would be for most users, since the attribute is currently completely undocumented AFAIK

But the use cases for the new addition are all people who are trying to use `__orig_bases__`! They should be able to find this through searching.

Reply:
Hah, fair point :-)